avaframe/com4FlowPy/flowCore.py

```diff
@@ -408,13 +409,12 @@ def calculation(args):
         row_idx = row_list[startcell_idx]
         col_idx = col_list[startcell_idx]
         dem_ng = dem[row_idx - 1: row_idx + 2, col_idx - 1: col_idx + 2]  # neighbourhood DEM
-        if varUmaxBool:
+        if varUmaxBool and varUmaxArray is not None:  # the check if uMax > 0 is already done where u is calculated to zDelta
             max_z_delta = varUmaxArray[row_idx, col_idx]
-        if varAlphaBool:
+        if varAlphaBool and varAlphaArray[row_idx, col_idx] > 0:
```

Comment:
in line 412 you explicitly check if ```varMaxArray is not None``` - for ```varAlphaBool``` and ```varExponentBool``` you don't explicitly check this (it only works for ```varExponentArray/varAlphaArray is None``` because ```if varAlphaBool/varExponentbool``` is evaluated first in the condition ... )--> maybe for easier readability you could rewrite to sth. like:

```
if varAlphaBool and varAlphaArray is not None:
    if varAlphaArray[row_idx, col_idx] > 0 and varAlphaArray[row_idx, col_idx] <= 90:
        alpha = varAlphaArray[row_idx, col_idx]

if varExponentBool and varExponentArray is not None:
    if var varExponentArray[row_idx, col_idx] > 0:
            exp = varExponentArray[row_idx, col_idx]
```